src/hooks/lib/preview/compounder/useCurvePreviewDepositFallback.ts

```diff
@@ -0,0 +1,57 @@
+import { Address, useContractRead } from "wagmi"
+
+import { VaultType } from "@/lib/types"
+import useCurvePreviewDepositUnderlying from "@/hooks/lib/preview/useCurvePreviewDepositUnderlying"
+import { useVaultContract } from "@/hooks/lib/useVaultContract"
+
+export default function useCurvePreviewDepositFallback({
+  asset,
+  vaultAddress,
+  token,
+  amount,
+  type,
+  slippage,
+  enabled,
+}: {
+  asset: Address
+  vaultAddress: Address
+  token?: Address
+  amount: string
+  type: VaultType
+  slippage: number
+  enabled?: boolean
+}) {
+  const isUnderlyingAsset = token !== asset
+  const preview = useContractRead({
+    ...useVaultContract(vaultAddress),
+    enabled: !!asset && !isUnderlyingAsset && enabled,
+    functionName: "previewDeposit",
+    args: [BigInt(amount)],
+  })
+
+  const previewUnderlying = useCurvePreviewDepositUnderlying({
+    asset,
+    vaultAddress,
+    token,
+    amount,
+    type,
+    slippage,
+    enabled: !!asset && isUnderlyingAsset && enabled,
+  })
+
+  return isUnderlyingAsset
+    ? {
+        ...previewUnderlying,
+        data: {
+          minAmountWei: BigInt(previewUnderlying?.data.minAmountWei).toString(),
```

Comment:
This is going to throw an error if `previewUnderlying.data.minAmountWei` is `undefined`. That result also should be a `BigInt` already, so you don't need to convert it to one manually.

You can just do: 

```ts
minAmountWei: previewUnderlying.data?.minAmountWei.toString()
```